amazon-kinesis-client/src/test/java/software/amazon/kinesis/coordinator/SchedulerTest.java

```diff
@@ -974,6 +958,54 @@ public void testErrorHandlerForUndeliverableAsyncTaskExceptions() {
         verify(rejectedTaskEvent, times(1)).accept(any());
     }
 
+    @Test
+    public void testUpdateStreamMapIfMissingLatestStream() throws Exception {
+        prepareMultiStreamScheduler(createDummyStreamConfigList(1, 6));
+        scheduler.checkAndSyncStreamShardsAndLeases();
+        verify(scheduler).syncStreamsFromLeaseTableOnAppInit(any());
+    }
+
+    @Test
+    public void testNotUpdatingStreamMapAsItContainsAllStreams() throws Exception {
+        final List<StreamConfig> streamConfigList = createDummyStreamConfigList(1, 6);
+        prepareMultiStreamScheduler(streamConfigList);
+        // Populate currentStreamConfigMap to simulate that the leader has the latest streams.
+        streamConfigList.forEach(s -> scheduler.currentStreamConfigMap().put(s.streamIdentifier(), s));
+        scheduler.checkAndSyncStreamShardsAndLeases();
+        verify(scheduler, never()).syncStreamsFromLeaseTableOnAppInit(any());
+    }
+
+    @SafeVarargs
+    private final void prepareMultiStreamScheduler(List<StreamConfig>... streamConfigs) {
```

Comment:
FYI: `private final` can be simplified to `private` (and some IDEs will flag this). Private methods are not visible for subclasses to override.

Reply:
My IDE doesn't not like the removal of the `final` keyword. `@SafeVarargs is not allowed on non-final instance methods`